### ISSUE TITLE
Crash when trying to create a shared link or opening app

### ISSUE THREAD

**Issue by CarlosCastaneda** [nextcloud/android#12649]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

Open App and Trier to create a public link for a folder

### Expected behaviour

Used to work in previous versions 

### Actual behaviour

Crash since version 3.28.0

### Android version

13

### Device brand and model

Samsung GalaxyA52 

### Stock or custom OS?

Stock

### Nextcloud android app version

3.28.0

### Nextcloud server version

28.0.2.5

### Using a reverse proxy?

No

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.util.Stack.push(java.lang.Object)' on a null object reference
    at com.owncloud.android.ui.activity.FileDisplayActivity.setSortListGroup(FileDisplayActivity.java:2496)
    at com.owncloud.android.ui.activity.FileDisplayActivity.setLeftFragment(FileDisplayActivity.java:616)
    at com.owncloud.android.ui.activity.FileDisplayActivity.showDetails(FileDisplayActivity.java:1524)
    at com.owncloud.android.ui.fragment.OCFileListFragment.onShareIconClick(OCFileListFragment.java:593)
    at com.owncloud.android.ui.adapter.OCFileListDelegate.showShareIcon$lambda$9(OCFileListDelegate.kt:396)
    at com.owncloud.android.ui.adapter.OCFileListDelegate.$r8$lambda$U1pswl3cp6slQ15jriz57LIrTJw(Unknown Source:0)
    at com.owncloud.android.ui.adapter.OCFileListDelegate$$ExternalSyntheticLambda3.onClick(Unknown Source:4)
    at android.view.View.performClick(View.java:7892)
    at android.view.View.performClickInternal(View.java:7869)
    at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
    at android.view.View$PerformClick.run(View.java:30891)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8762)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a52q`
* Model: `SM-A525F`
* Id: `TP1A.220624.014`
* Product: `a52qnseea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `A525FXXS6DWK2`

**Comment by CarlosCastaneda** [comment]
After opening the App:

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.util.Stack.push(java.lang.Object)' on a null object reference
    at com.owncloud.android.ui.activity.FileDisplayActivity.setSortListGroup(FileDisplayActivity.java:2496)
    at com.owncloud.android.ui.activity.FileDisplayActivity.setLeftFragment(FileDisplayActivity.java:616)
    at com.owncloud.android.ui.activity.FileDisplayActivity.setLeftFragment(FileDisplayActivity.java:604)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onNewIntent(FileDisplayActivity.java:571)
    at android.app.Activity.performNewIntent(Activity.java:8962)
    at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1556)
    at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1569)
    at android.app.ActivityThread.deliverNewIntents(ActivityThread.java:4362)
    at android.app.ActivityThread.handleNewIntent(ActivityThread.java:4369)
    at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:56)
    at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:230)
    at android.os.Looper.loop(Looper.java:319)
    at android.app.ActivityThread.main(ActivityThread.java:8893)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a52q`
* Model: `SM-A525F`
* Id: `UP1A.231005.007`
* Product: `a52qnseea`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `A525FXXU6EXB3`

**Comment by milgner** [comment]
Got a similar one, except with `Stack.pop`

```
### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.util.Stack.pop()' on a null object reference
    at com.owncloud.android.ui.activity.FileDisplayActivity.popSortListGroupVisibility(FileDisplayActivity.java:2505)
    at com.owncloud.android.ui.activity.FileDisplayActivity.popBack(FileDisplayActivity.java:1048)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onBackPressed(FileDisplayActivity.java:1003)
    at com.owncloud.android.ui.activity.FileDisplayActivity.onOptionsItemSelected(FileDisplayActivity.java:823)
    at android.app.Activity.onMenuItemSelected(Activity.java:4421)
    at androidx.activity.ComponentActivity.onMenuItemSelected(ComponentActivity.java:528)
    at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:264)
    at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:269)
    at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:110)
    at androidx.appcompat.widget.ToolbarWidgetWrapper$1.onClick(ToolbarWidgetWrapper.java:188)
    at android.view.View.performClick(View.java:7515)
    at android.view.View.performClickInternal(View.java:7492)
    at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
    at android.view.View$PerformClick.run(View.java:29356)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:8163)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:946)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280090`
* Build flavor: `gplay`

### Device information
* Brand: `Lenovo`
* Device: `TB128FU`
* Model: `Lenovo TB128FU`
* Id: `TKQ1.221220.001`
* Product: `TB128FU_EEA`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `TB128FU_USR_S100025_2401172249_Q00126_ROW`
```

**Comment by joshtrichards** [comment]
Fixed in #12584
